src/main/java/com/softteam/vocabuilder/controller/CategoryController.java

```diff
@@ -0,0 +1,47 @@
+package com.softteam.vocabuilder.controller;
+
+import com.softteam.vocabuilder.persistence.entity.Category;
+import com.softteam.vocabuilder.service.CategoryServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.MapsId;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/categories")
+
+public class CategoryController {
+
+    @Autowired
+    private CategoryServiceImpl categoryService;
+    @PostMapping
+    public ResponseEntity<Category> createCategory(@RequestBody Category category){
+        categoryService.create(category);
+        return new ResponseEntity<Category>(HttpStatus.CREATED);
+    }
+    @PatchMapping("/update/{id}")
```

Comment:
we need to use PUT as the default update method, PATH is used in some specific cases
also you don't need to specify the word `update` in the path, the HTTP Method is for that

Reply:
Annotated and resolve.